src/Adapters/Laravel/Commands/TestCommand.php

```diff
@@ -132,20 +133,67 @@ protected function phpunitArguments($options)
     protected function clearEnv()
     {
         if (!$this->option('env')) {
-            $repositories = RepositoryBuilder::create()
-                ->make();
-
-            $envs = Dotenv::create(
-                $repositories,
+            $vars = self::getEnvironmentVariables(
                 // @phpstan-ignore-next-line
                 $this->laravel->environmentPath(),
                 // @phpstan-ignore-next-line
                 $this->laravel->environmentFile()
-            )->safeLoad();
+            );
+
+            $repository = Env::getRepository();
```

Comment:
This is an important bug fix. The old implementation was bad for two reasons:

1. It was slow because it loaded everything again and then cleared it.
2. It was incorrect because it didn't respect the fact that Laravel uses "immutable" mode when loading .env files which means that if a variable is already defined, the entry in the .env file is effectively skipped, but the old code in this package used to blanket delete everything.

Reply:
LOL, just started working on this 5 mins ago. Thanks!

Reply:
Ok, so I need to do a PR for the laravel 7 version to fix the bug, and to the laravel 8 version to upgrade to phpdotenv v5?

Reply:
For the Laravel 7, you should pull request against: https://github.com/nunomaduro/collision/tree/v4.x.